macros/dedupe_artifacts_v1.sql

```diff
@@ -0,0 +1,24 @@
+{% macro dedupe_dbt_artifacts_v1() %}
+
+{% set src_dbt_artifacts = source('dbt_artifacts', 'artifacts') %}
+
+{% set dedupe_results_query %}
+
+    create temporary table {{ src_dbt_artifacts.database }}.{{ src_dbt_artifacts.schema }}.dbt_temp_artifact_table as
+        select distinct * from {{ src_dbt_artifacts }};

```

Comment:
This is potentially a very expensive operation as it is effectively a group by all columns. If we know that one column represents the uniqueness of a record correctly, we can just do a:
```
select * from {{ src_dbt_artifacts }}
qualify row_number() over (partition by invocation_id order by generated_at asc) = 1
```

which is a lot faster

Reply:
Good idea. The `generated_at` sort is a bit redundant - because it will be the same for each artifact that's duplicated given it's a property of the artifact not the upload time, but the rest is good.

Reply:
... in fact. I think the best thing to quantify _uniqueness_ is probably the timestamp itself. Is that a crazy idea?

I'm thinking that in the case of a dbt cloud run, where we want to group together multiple results into the same run, the artifact_run_id isn't unique - there may be several different artifacts with the same id. But the timestamp shows that they're different results. Ditto the V1 artifacts.

Reply:
I don't see why that should be an issue given that the precision is microseconds. I'm in support!

Reply:
I've totally been overthinking this - see #103 . It should totally be `command_invocation_id` here.

I've update PR and should be ready to rock.